pkg/utils/managedresources/managedresources_test.go

```diff
@@ -559,7 +559,7 @@ var _ = Describe("managedresources", func() {
 			timeoutCtx, cancel := context.WithTimeout(ctx, time.Second*5)
 			defer cancel()
 			Expect(WaitUntilHealthy(timeoutCtx, fakeClient, namespace, name)).To(Succeed())
-			Expect(shouldFail.Load()).To(BeFalse()) // to ensure that the goroutine actually applies all patches
+			Eventually(shouldFail.Load()).Should(BeFalse()) // to ensure that the goroutine actually applies all patches
```

Comment:
I was also thinking about such fix but I don't think it is super correct one. The whole purpose of `shouldFail` is to prevent `WaitUntilHealthy` to succeed before the goroutine did all updates. With the above change, there could be a bug where `WaitUntilHealthy` succeeds right away wrongly, then due to the `Eventually` usage the assertion for shouldFail passes after several retries.

Reply:
Yes, I see. But this was already the case even with the previous implementation. In case `WaitUntilHealthy ` succeeds immediately then the race is between the 
`Expect(shouldFail.Load())` & all the updates of the MR status + `shouldFail.Store(false)`

Do you have a suggestion how can we prevent that? My thinking is that if we want to test that all other status combinations are not healthy then we should have tests where the wait function fails because of that